r/R/array.R

```diff
@@ -84,6 +84,28 @@
 #'
 #' @rdname array
 #' @name array
+#' @examples
+#' vals <- 1:10
+#' my_array <- Array$create(vals)
+#' my_array$type
+#' my_array$cast(int8())
+#' 
+#' # Check if value is null; zero-indexed
+#' na_array <- Array$create(c(1:5, NA))
+#' na_array$IsNull(0)
+#' na_array$IsNull(5)
+#' na_array$IsValid(5)
+#' na_array$null_count
+#' 
+#' # zero-copy slicing; the offset of the new Array will be the same as the index passed to $Slice
```

Comment:
This is probably (definitely) not the right place for it, and feel free to punt on this, but it would be nice to have a description of why one might want zero-copy and what the benefits of it are.

Reply:
Ooh, reckon that would fit nicely in one of the vignettes?